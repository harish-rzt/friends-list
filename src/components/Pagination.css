:local(.wrapper) {
  background-color: #ffffff;
  padding: 8px 0;
  display: flex;
  justify-content: center;
  align-items: center;
}

:local(.btnAction) {

}

:local(.info) {
  padding: 0 20px;
  font-size: 15px;
}
