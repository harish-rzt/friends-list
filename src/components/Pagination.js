import classnames from 'classnames';
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { decrementPagination, incrementPagination } from '../actions/FriendsActions';
import styles from './Pagination.css';


class Pagination extends Component {
  render() {
    const {
      friendlist: { pagination: { currentPage, totalPages } },
      decrementPagination,
      incrementPagination,
    } = this.props;
    return (
      <div className={styles.wrapper}>
        <button
          className={`btn btn-default ${styles.btnAction}`}
          onClick={() => decrementPagination()}
          disabled={currentPage <= 1}
        >
          <i className={classnames('fa', 'fa-chevron-left')} />
        </button>
        <span className={styles.info}>
          {currentPage} / {totalPages} pages
        </span>
        <button
          className={`btn btn-default ${styles.btnAction}`}
          onClick={() => incrementPagination()}
          disabled={currentPage >= totalPages}
        >
          <i className={classnames('fa', 'fa-chevron-right')} />
        </button>
      </div>
    );
  }
}

Pagination.propTypes = {};

function mapStateToProps(state) {
  return state;
}

export default connect(mapStateToProps, {
  incrementPagination,
  decrementPagination,
})(Pagination);