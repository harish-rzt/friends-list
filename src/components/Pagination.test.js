import Pagination from './Pagination';

const pagination = { currentPage: 1, totalPages: 2 };

describe('<Pagination />', () => {
  it('Render check with shallow', () => {
    const wrapper = shallow(<Pagination
      pagination={pagination}
      incrementPagination={f => f}
      decrementPagination={f => f}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Render check with render', () => {
    const wrapper = render(<Pagination
      pagination={pagination}
      incrementPagination={f => f}
      decrementPagination={f => f}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('DOM structure test', () => {
    const wrapper = mount(<Pagination
      pagination={pagination}
      incrementPagination={f => f}
      decrementPagination={f => f}
    />);

    // Testing the DOM structure with two buttons(next & previous)
    const buttonsCount = wrapper.find("button").length;
    expect(buttonsCount).toEqual(2);

    // Testing the rendered pagination text with passed prop
    const paginationInfoText = wrapper.find("span").text();
    expect(paginationInfoText).toEqual("1 / 2 pages");
  });

  it('Simulating click events', () => {
    const incrementSpy = sinon.spy();
    const decrementSpy = sinon.spy();
    const wrapper = shallow(<Pagination
      pagination={pagination}
      decrementPagination={decrementSpy}
      incrementPagination={incrementSpy}
    />);

    // simulate click on left arrow click(decrement pagination),
    wrapper.find("button").first().simulate("click");
    expect(decrementSpy.calledOnce).toBe(true);

    // simulate click on right arrow click(increment pagination)
    wrapper.find("button").at(1).simulate("click");
    expect(incrementSpy.calledOnce).toBe(true);
  });
});