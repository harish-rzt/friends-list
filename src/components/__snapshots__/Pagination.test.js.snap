exports[`<Pagination /> Render check with render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "btn btn-default undefined",
          "disabled": "",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "fa fa-chevron-left",
            },
            "children": Array [],
            "name": "i",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "1 / 2 pages",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "btn btn-default undefined",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "fa fa-chevron-right",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "disabled": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "disabled": undefined,
        },
      },
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "1 / 2 pages",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "btn btn-default undefined",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "fa fa-chevron-right",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "btn btn-default undefined",
            "disabled": "",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "fa fa-chevron-left",
              },
              "children": Array [],
              "name": "i",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "disabled": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "disabled": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "class": "btn btn-default undefined",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "fa fa-chevron-right",
            },
            "children": Array [],
            "name": "i",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "1 / 2 pages",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "btn btn-default undefined",
              "disabled": "",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "fa fa-chevron-left",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "disabled": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "disabled": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`<Pagination /> Render check with shallow 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    decrementPagination={[Function]}
    incrementPagination={[Function]}
    pagination={
        Object {
            "currentPage": 1,
            "totalPages": 2,
          }
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn btn-default undefined"
          disabled={true}
          onClick={[Function]}>
          <i
                    className="fa fa-chevron-left" />
</button>,
        <span
          className={undefined}>
          1
           / 
          2
           pages
</span>,
        <button
          className="btn btn-default undefined"
          disabled={false}
          onClick={[Function]}>
          <i
                    className="fa fa-chevron-right" />
</button>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa fa-chevron-left" />,
          "className": "btn btn-default undefined",
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-chevron-left",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1,
            " / ",
            2,
            " pages",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          1,
          " / ",
          2,
          " pages",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa fa-chevron-right" />,
          "className": "btn btn-default undefined",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-chevron-right",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn btn-default undefined"
            disabled={true}
            onClick={[Function]}>
            <i
                        className="fa fa-chevron-left" />
</button>,
          <span
            className={undefined}>
            1
             / 
            2
             pages
</span>,
          <button
            className="btn btn-default undefined"
            disabled={false}
            onClick={[Function]}>
            <i
                        className="fa fa-chevron-right" />
</button>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-chevron-left" />,
            "className": "btn btn-default undefined",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-chevron-left",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1,
              " / ",
              2,
              " pages",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            1,
            " / ",
            2,
            " pages",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-chevron-right" />,
            "className": "btn btn-default undefined",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-chevron-right",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
